marlowe/src/Language/Marlowe/Scripts.hs

```diff
@@ -171,6 +179,122 @@ mkValidator :: MarloweParams -> Scripts.ValidatorType MarloweStateMachine
 mkValidator p = SM.mkValidator $ SM.mkStateMachine Nothing (mkMarloweStateMachineTransition p) isFinal
 
 
+{-# INLINABLE smallMarloweValidator #-}
+smallMarloweValidator
+    :: MarloweParams
+    -> MarloweData
+    -> [Input]
+    -> ScriptContext
+    -> Bool
+smallMarloweValidator MarloweParams{rolesCurrency, rolePayoutValidatorHash} MarloweData{..} inputs ctx@ScriptContext{scriptContextTxInfo} = do
+    let slotConfig = def :: TimeSlot.SlotConfig
+    let ownInput = case findOwnInput ctx of
+            Just i -> i
+            _      -> traceError "I0" {-"Can't find validation input"-}
+    let scriptInValue = txOutValue $ txInInfoResolved ownInput
+    let (minTime, maxTime) =
+            case txInfoValidRange scriptContextTxInfo of
+                Interval.Interval (Interval.LowerBound (Interval.Finite l) True) (Interval.UpperBound (Interval.Finite h) False) -> (l, h)
```

Comment:
FYI: I double-checked in the [ledger spec](https://hydra.iohk.io/build/7840126/download/1/shelley-ma.pdf) that the `[closed, open)` interpretation here of the validity interval is correct.